src/foam/dao/AbstractDAO.java

```diff
@@ -58,6 +58,8 @@ protected Sink decorateSink_(Sink sink, long skip, long limit, Comparator order,
       sink = new PredicatedSink(predicate, sink);
     }
 
+    ((AbstractFObject) sink).setX(x);
```

Comment:
I see.

I feel like we should be using x.create() in all those places rather than new Foo(), but I will get caught up on that code to see if we did it this way for any particular reason.

In this case in particular, setX() will only set the context on the latest decorated sink, but if we did a skipped, ordered, predicated sink, the skip and ordered Sinks will not have their context set.  I'm okay if we leave it as new Foo() but add the setX() calls after each new Foo() statement.